packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -0,0 +1,244 @@
+    const clickedCell = event.target;
+    clickedCell.setAttribute('tabIndex', '0');
+    clickedCell.focus();
+    event.preventDefault();
+  }
+
+  getNumberOfHeaderRows() {
+    return this.props.columnHeaders?.pinnedColumns?.length > 0 ? 1 : 0;
+  }
+
+  getCellData(cell, cellColumnIndex) {
+    const cellClassName = { className: classNames(cx(this.props.rowHeaderIndex === cellColumnIndex ? 'worklist-data-grid-row-header' : 'worklist-data-grid-cell-data')) };
+    const tabIndex = { tabIndex: '-1' };
+    return this.props.rowHeaderIndex === cellColumnIndex ? (<th key={cellColumnIndex} {...tabIndex} {...cellClassName} role="rowheader">{cell.cellContent}</th>) : (<td key={cellColumnIndex} {...tabIndex} {...cellClassName}>{cell.cellContent}</td>);
+  }
+
+  buildColumnHeaders(colHeader) {
+    if (colHeader?.pinnedColumns?.length > 0) {
+      return (
+        <tr height={this.props.columnHeaderHeight}>
+          {colHeader.pinnedColumns.map(columnData => (this.buildSingleColumnHeading(columnData)))}
+        </tr>
+      );
+    }
+    return undefined;
+  }
+
+  buildSingleColumnHeading(columnData) {
+    const width = columnData.width || this.props.columnHeaderDefaultWidth;
+    const height = columnData.height || this.props.columnHeaderHeight;
+    const columnHeaderClassName = classNames(cx('worklist-data-grid-column-header'));
+    return (
+      /* eslint-disable react/forbid-dom-props */
+      <th key={columnData.id} id={columnData.id} role="columnheader" className={columnHeaderClassName} tabIndex="-1" style={{ width, height }}>{columnData.displayName}</th>
+    );
+  }
+
+  buildSingleRow(row, rowIndex) {
+    const backgroundClassName = classNames(cx({ striped: !!(rowIndex % 2) }));
+    return (
+      <tr key={rowIndex} className={backgroundClassName} height={row.height || this.props.rowDefaultHeight}>
+        {row.cells.map((cell, cellColumnIndex) => (
+          this.getCellData(cell, cellColumnIndex)
+        ))}
+      </tr>
+    );
+  }
+
+  buildRows(rows) {
+    return (
+      rows.map((element, index) => (this.buildSingleRow(element, index)))
+    );
+  }
+
+  render() {
+    const {
+      id,
+      ariaLabelledby,
+      ariaLabel,
+      columnHeaders,
+      rows,
+      columnHeaderDefaultWidth,
+      ...customProps
+    } = this.props;
+
+    const theme = this.context;
+
+    const gridClassNames = classNames(cx('worklist-data-grid', theme.className), customProps.className);
+
+    return (
+      <div className={cx(['container'])}>
+        <table
+          id={id}
+          role="grid"
+          aria-labelledby={ariaLabelledby}
+          aria-label={ariaLabel}
+          className={gridClassNames}
+        >
+          <tbody>
+            {this.buildColumnHeaders(columnHeaders)}
+            {this.buildRows(rows)}
+          </tbody>
+        </table>
+      </div>
+    );
+  }
+}
+
+WorklistDataGrid.propTypes = propTypes;
+WorklistDataGrid.contextType = ThemeContext;
```

Comment:
We would want to use a functional component so that we are not locked in to a single context for this component.